R/get_plot_forecast_data.R

```diff
@@ -0,0 +1,75 @@
+#' Combines load_truth() and pivot_forecasts_wider(), returns the data.
+#'
+#' @param data forecasts data frame from load_forecast(). It has columns model, 
+#' forecast_date,location, target, type, quantile, value, horizon and 
+#' target_end_date.
+#' @param model_to_plot characters of model abbreviations 
+#' @param horizons_to_plot forecasts are plotted for the horizon time steps after the 
+#' forecast date
+#' @param quantiles_to_plot vector of quanitles to include in the plot
+#' @param location_to_plot character vector of one location. 
+#' @param truth_source character vector specifying where the truths will
+#' be loaded from: currently support "JHU","USAFacts", "NYTimes"
+#' @param target_variable string specifying target type. It should be one of 
+#' "Cumulative Cases","Cumulative Deaths","Incident Cases" and "Incident Deaths."
+#' @param  truth_as_of the plot includes the truth data that would have been 
+#' in real time as of the truth_as_of date.
+#' 
+#' @return data frame with columns model, 
+#' forecast_date, location, inc_cum, death_case, type, quantile, value, horizon and 
+#' target_end_date.
+#' 
+#' @export
+get_plot_forecast_data <- function(data, 
+                                   model_to_plot,
+                                   horizons_to_plot,
+                                   quantiles_to_plot,
+                                   location_to_plot,
+                                   truth_source,
+                                   target_variable,
+                                   truth_as_of = NULL){
```

Comment:
it looks like truth_as_of is not used below.  Let's add an issue for this and issue a warning if the user provides a truth_as_of argument.

Reply:
Since `truth_as_of` is not yet supported, in practice in the near term I would probably use my own truth loaded from covidData.  Can we have an option here to *not* load any truth data so that I can add it manually until the functionality for versioned truth exists?